packages/manager/modules/pci/src/projects/project/kubernetes/add/add.controller.js

```diff
@@ -60,12 +61,16 @@ export default class {
   create() {
     this.sendKubeTrack('add::confirm');
 
+    const { nodePool } = this.cluster;
     this.isAdding = true;
     const options = {
-      desiredNodes: this.cluster.nodePool.nodeCount,
-      flavorName: this.cluster.nodePool.flavor.name,
-      antiAffinity: this.cluster.nodePool.antiAffinity,
-      monthlyBilled: this.cluster.nodePool.monthlyBilling,
+      desiredNodes: nodePool.nodeCount,
+      flavorName: nodePool.flavor.name,
+      antiAffinity: nodePool.antiAffinity,
+      monthlyBilled: nodePool.monthlyBilling,
+      autoscale: nodePool.autoscaling.autoscale,
+      minNodes: nodePool.autoscaling.nodes.lowest.value,
+      maxNodes: nodePool.autoscaling.nodes.highest.value,
```

Comment:
Wouldn't it be easier to use a model that would abstract all the nesting ?

Reply:
Iam not sure we can make a model an just send it like it.
for example my `autoscaling` model can't be just composed of `minNodes`, `maxNodes` ...

So for me no.

Reply:
I was thinking of a model for `NodePool` that way we could just nodePool.maxNodes and it would abstract all the nesting

Reply:
yes i understand @marie-j.
my `autoscaling` model is deeply composed.

You can see it here:
https://github.com/ovh/manager/blob/607cde76019fdfe2ff07913df3fa6ed2f625d95c/packages/manager/modules/pci/src/projects/project/kubernetes/kubernetes.routing.js#L57

Reply:
I agree with marie. Maybe we can create a nodepool and autoscaling class and initiate it with all this info and pass it to service/api. The autoscaling class can be a composite class in nodepool class. You can move all the complex logic of creating an autoscaling object into these classes. This way it looks clean, easy to understand, and can scale well with further enhancements. Just my thoughts :)

Reply:
Why not for a classes.
I will create a Jira task to make refactor cross Kube modules.

Thanks for suggestion.

Reply:
We already have a NodePool class which give us the nodePool info (scale, minNodes, maxNodes, ...).
In my case i created a model to manage 3 Numeric values (lowest, desired and highest).
All the logic for the autosacling is implemented into the AutoscalingController.